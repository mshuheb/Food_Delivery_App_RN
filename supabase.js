import { createClient } from "@supabase/supabase-js";
import "react-native-url-polyfill"

const supabaseUrl= "https://xkveywmtbhzcgmdvtvjq.supabase.co"

const supabaseAnonkey = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InhrdmV5d210Ymh6Y2dtZHZ0dmpxIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MzAzODQ1ODEsImV4cCI6MjA0NTk2MDU4MX0.JE8CA1kGKhzcYXZRoo0AEzu5NJ4nmAU2qcpQYSgQ1qw"

export const supabase =  createClient(supabaseUrl, supabaseAnonkey)