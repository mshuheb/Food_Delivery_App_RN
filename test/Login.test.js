import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { Alert } from "react-native"; // Ensure Alert is imported to be mocked
import Login from "../app/(authenticate)/login";
import { supabase } from "../supabase";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { useRouter } from "expo-router";

// Mock external modules
jest.mock("../supabase");
jest.mock('@react-native-async-storage/async-storage', () => ({
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
}));
jest.mock("expo-router", () => ({
  useRouter: jest.fn(),
}));

// Mock expo-vector-icons
jest.mock('@expo/vector-icons', () => ({
  AntDesign: 'AntDesign',  // Mocking icon
  MaterialIcons: 'MaterialIcons',  // Mocking icon
  FontAwesome: 'FontAwesome',  // Mocking icon
}));

// Mock Alert
jest.spyOn(Alert, 'alert').mockImplementation(() => {});

describe("<Login />", () => {
  let routerMock;

  beforeEach(() => {
    routerMock = { replace: jest.fn() };
    useRouter.mockReturnValue(routerMock);
    AsyncStorage.getItem.mockResolvedValue(null);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly with initial UI elements", () => {
    const { getByPlaceholderText, getByText, toJSON } = render(<Login />);

    expect(getByText("Food App")).toBeTruthy();
    expect(getByText("Log in to your account")).toBeTruthy();
    expect(getByPlaceholderText("Enter your email")).toBeTruthy();
    expect(getByPlaceholderText("Enter your password")).toBeTruthy();
    expect(getByText("Login")).toBeTruthy();
    expect(getByText("Don't have an account? Sign Up")).toBeTruthy();

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  test("allows user to input email and password", () => {
    const { getByPlaceholderText, toJSON } = render(<Login />);

    const emailInput = getByPlaceholderText("Enter your email");
    const passwordInput = getByPlaceholderText("Enter your password");

    fireEvent.changeText(emailInput, "test@example.com");
    fireEvent.changeText(passwordInput, "password123");

    expect(emailInput.props.value).toBe("test@example.com");
    expect(passwordInput.props.value).toBe("password123");

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  test("shows alert on login error", async () => {
    const { getByText, getByPlaceholderText, toJSON } = render(<Login />);

    supabase.auth.signInWithPassword.mockResolvedValue({
      error: { message: "Invalid email or password" },
      data: null,
    });

    const emailInput = getByPlaceholderText("Enter your email");
    const passwordInput = getByPlaceholderText("Enter your password");
    const loginButton = getByText("Login");

    fireEvent.changeText(emailInput, "wrong@example.com");
    fireEvent.changeText(passwordInput, "wrongpassword");
    fireEvent.press(loginButton);

    await waitFor(() => {
      expect(supabase.auth.signInWithPassword).toHaveBeenCalledWith({
        email: "wrong@example.com",
        password: "wrongpassword",
      });
      expect(Alert.alert).toHaveBeenCalledWith("Invalid email or password. Please try again.");
    });

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  test("navigates to home screen on successful login", async () => {
    supabase.auth.signInWithPassword.mockResolvedValue({
      error: null,
      data: { session: { access_token: "test_token" } },
    });

    const { getByText, getByPlaceholderText, toJSON } = render(<Login />);
    const emailInput = getByPlaceholderText("Enter your email");
    const passwordInput = getByPlaceholderText("Enter your password");
    const loginButton = getByText("Login");

    fireEvent.changeText(emailInput, "test@example.com");
    fireEvent.changeText(passwordInput, "password123");
    fireEvent.press(loginButton);

    await waitFor(() => {
      expect(supabase.auth.signInWithPassword).toHaveBeenCalledWith({
        email: "test@example.com",
        password: "password123",
      });
      expect(AsyncStorage.setItem).toHaveBeenCalledWith("authToken", "test_token");
      expect(AsyncStorage.setItem).toHaveBeenCalledWith("userEmail", "test@example.com");
      expect(routerMock.replace).toHaveBeenCalledWith("/(home)");
    });

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  test("redirects to home if user is already logged in", async () => {
    AsyncStorage.getItem.mockResolvedValue("existing_token");

    const { toJSON } = render(<Login />);

    await waitFor(() => {
      expect(routerMock.replace).toHaveBeenCalledWith("/(home)");
    });

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  test("redirects to sign-up screen on 'Sign Up' press", () => {
    const { getByText, toJSON } = render(<Login />);
    const signUpText = getByText("Don't have an account? Sign Up");

    fireEvent.press(signUpText);

    expect(routerMock.replace).toHaveBeenCalledWith("/register");

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });
});
