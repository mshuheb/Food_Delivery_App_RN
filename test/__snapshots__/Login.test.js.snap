// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> allows user to input email and password 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Log in to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Enter your email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="test@example.com"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="password123"
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 12,
        }
      }
    >
      <Text>
        Keep me Logged In
      </Text>
      <Text>
        Forgot Password
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Don't have an account? Sign Up
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<Login /> navigates to home screen on successful login 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Log in to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Enter your email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="test@example.com"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="password123"
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 12,
        }
      }
    >
      <Text>
        Keep me Logged In
      </Text>
      <Text>
        Forgot Password
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Don't have an account? Sign Up
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<Login /> redirects to home if user is already logged in 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Log in to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Enter your email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 12,
        }
      }
    >
      <Text>
        Keep me Logged In
      </Text>
      <Text>
        Forgot Password
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Don't have an account? Sign Up
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<Login /> redirects to sign-up screen on 'Sign Up' press 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Log in to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Enter your email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 12,
        }
      }
    >
      <Text>
        Keep me Logged In
      </Text>
      <Text>
        Forgot Password
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Don't have an account? Sign Up
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<Login /> renders correctly with initial UI elements 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Log in to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Enter your email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 12,
        }
      }
    >
      <Text>
        Keep me Logged In
      </Text>
      <Text>
        Forgot Password
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Don't have an account? Sign Up
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<Login /> shows alert on login error 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Log in to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Enter your email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="wrong@example.com"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="wrongpassword"
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 12,
        }
      }
    >
      <Text>
        Keep me Logged In
      </Text>
      <Text>
        Forgot Password
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Don't have an account? Sign Up
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
